CONTRIBUTING.md

```diff
@@ -0,0 +1,231 @@
+
+#### Tokens for a specific language
+
+Perhaps you want a particular token to be styled differently in a certain language. For example, CSS uses `.important` to style `!important`, while Markdown uses the same token in its headings, and you want them to have different styles. You can achieve it with the following selectors:
+
+```css
+/* Tokens with the `.important` class */
+.token.important {
+	/* styles */
+}
+
+/* Tokens in a CSS block with the `.important` class */
+.language-css .token.important {
+	/* styles */
+}
+
+/* Tokens in a Markdown block with the `.important` class */
+.language-markdown .token.important {
+	/* styles */
+}
+```
+
+### Code style guidelines
+
+The most important one is that Prism uses [tabs for indentation and spaces for alignment](https://lea.verou.me/2012/01/why-tabs-are-clearly-superior/). For the rest, just run `npm run lint` to check (and `npm run lint-fix` to fix)!
+
+### Theme guidelines and requirements
+
+Your theme will be used by Prism and its plugins, which have default CSS rules of their own. To ensure that your theme is compatible with Prism and its plugins, please follow these guidelines!
+
+#### Do not use CSS variables, at least not yet
+
+This may be a little strange, but it's because Prism supports IE11, which does not support CSS variables. However, we will be [dropping support for IE11 in Prism V2](https://github.com/PrismJS/prism/issues/1578), so you are encouraged to leave commented-out CSS variables (or at least a list of the colours you used) in your CSS files for an easier transition later on!
+
+#### Ensure that your token selectors include the `.token` class
+
+Prism adds the class `.token` to the, well, tokens, that it highlights with its wizardry magic. As such, all tokens that are highlighted by Prism will include the `.token` class. When you declare a style for a specific token, please make sure to include the `.token` class in your selector!
+
+```css
+/* Do this */
+.token.token-name {
+	/* styles */
+}
+
+/* Do not do this */
+.token-name {
+	/* styles */
+}
+```
+
+#### Maintain a font size of `1em` for code blocks
+
+Some of Prism's plugins assume that the `pre` and `code` elements have the same font size, else there will be bugs pertaining to misaligned and so forth. As such, please set the `font-size` of `code[class*="language-"], pre[class*="language-"]` to `1em`, i.e.:
+
+```css
+code[class*="language-"],
+pre[class*="language-"] {
+	font-size: 1em;
+	/* more styles */
+}
+```
+
+#### Do not set the `overflow` property of `pre` elements
+
+The `overflow` property of `pre` elements must be left unset, or set to `auto`. This is because other `overflow` values will cause problems with layouts based on `float` or flexboxes.
+
+One notable exception is the original Coy theme because its shadows are impossible to implement otherwise.
```

Comment:
Hmm, I'm not so sure about that, since I did write about not using CSS variables (for V1) earlier. Though maybe I could just remove it anyway, since I added a comment in the code that `overflow: auto` is the standard.

Reply:
It feels weird mentioning Coy here. Coy is a pretty weird theme in more ways than one (which causes us a few problems).

You really shouldn't do the weird stuff Coy does, so I don't want someone reading this getting the impression "if Coy (a pretty popular theme) does it, then it's fine for me to do it too"!